distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1185,6 +1187,66 @@ public void testRetrievalReliablility() throws Exception {
     }
 
     /**
+     * Tests that if the endpoint disconnects twice, the retrieval retries both times
+     * This test will respond with the correct Partial Content when a range header is sent in the request
+     * @throws Exception
+     */
+    @Test
+    @Ignore
+    public void testRetrievalWithByteOffset() throws Exception {
+
+        cometDClient = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL, ACTIVITIES_CHANNEL));
+        String filename = "product2.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+        HeaderCapture headerCapture = new HeaderCapture();
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.custom(headerCapture))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData,
+                                Duration.ofMillis(200),
+                                2,
+                                headerCapture.getHeaders));
+
+        String restUrl = REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + metacardId
+                + "?transform=resource" + "&session=" + cometDClient.getClientId();
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is(resourceData));
+        // @formatter:on
+        cswServer.verifyHttp().times(3, Condition.uri("/services/csw"), Condition.parameter("request", "GetRecordById"));
+
+        expect("Waiting for notifications")
+                .within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(NOTIFICATIONS_CHANNEL).size() == 3);
+        expect("Waiting for activities")
+                .within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(ACTIVITIES_CHANNEL).size() == 7);
+
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(3));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)), filename, "Resource retrieval retrying after 1 bytes. Attempt 1 of 3.", "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(1)), filename, "Resource retrieval retrying after 1 bytes. Attempt 2 of 3.", "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(2)), filename, "Resource retrieval completed, 8 bytes retrieved. ", "complete");
```

Comment:
I had copied them from the testRetrievalReliability test since it should essentially be testing the same cometD events. It looks like we removed the cometD assertions from that test recently though, so I'll update these to reflect those changes.